src/BaseballGame.js

```diff
@@ -0,0 +1,84 @@
+//게임진행
+import {displayGameover, displayResult} from "./GameView.js"
+
+export default class BaseballGame{
+    constructor(){
+        this.computerNumbers = this.createNewNumbers();
+    }
+
+    createNewNumbers(){
+        let numbers = [];
+            while (numbers.length < 3) {
+                let num = MissionUtils.Random.pickNumberInRange(1,9);
+                if (!numbers.includes(num)) {
+                    numbers.push(num);
+                }
+            }
+            // let numbers = MissionUtils.Random.pickUniqueNumbersInRange(1, 9, 3);
+            numbers = numbers.join("");
+            console.log(numbers);
+
+        return numbers; //문자열
+    }
+    
+
+    gamePlay(userInput){
+        // console.log(userInput);
+        if(this.checkUserInput(userInput)){
+            return ;    //숫자 아니면 끝내기
+        }
+    
+        const result = this.compareInputWithAnswer(userInput);
+        
+        if(result==="3스트라이크"){
+            displayGameover();
+        }else {
+            // console.log(result);
+            displayResult(result);
+        }
+    }
+
+    compareInputWithAnswer(userInput){
+        let strikeCount=0, ballCount=0;
+    
+        for(let i=0;i<3;i++){
+            if(userInput[i] === this.computerNumbers[i]){  //스트라이크
+                strikeCount++;
+            }else if(this.computerNumbers.includes(userInput[i])){   //볼
+                ballCount++;
+            }
+        }
+    
+        const res = this.play(strikeCount, ballCount);
+        return res;
```

Comment:
앞선 리뷰에서 언급한 내용 이었지만 리팩터링하면서 변수명을 다시 생각해보면 좋을 것 같아요!
`res`처럼 축약된 형태의 네이밍 보다 더 명확한 의미를 담은 이름으로 바꾸는 건 어떨까요?

그리고 여기도 얼리 리턴을 적용할 수 있어 보입니다!

```suggestion
       return this.play(strikeCount, ballCount);
```